torch/nn/parallel/scatter_gather.py

```diff
@@ -57,6 +57,9 @@ def gather_map(outputs):
             return Gather.apply(target_device, dim, *outputs)
         if out is None:
             return None
+        if isinstance(out, dict):
+            return type(out)(((k, gather_map([d[k] for d in outputs]))
+                              for k in out))
```

Comment:
One extra thing to check (and raise an error otherwise): `all(len(d) == len(out) for d in outputs)` (to make sure the sets of keys are equal)